Added EmojiReaction payload types

Samyoul opened the pull request:
Added emoji reaction payload specifications.

For reference see:

- [`status-go` implementation](https://github.com/status-im/status-go/pull/1999)
- [`status-react` implementation](https://github.com/status-im/status-react/pull/10912)
- [UI process flows](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1239%3A0)

I've added these specs to the draft version of `6-payloads` because I saw images in drafts and presumed that emoji reactions would also be considered on the same level or lower of spec stability.

Samyoul committed:
Added EmojiReaction payload types

Samyoul committed:
Added EmojiReaction payload types

Samyoul committed:
I'm an author now

Samyoul committed:
wordlist update

Samyoul commented on docs/draft/6-payloads.md:
```diff
@@ -144,7 +150,7 @@ message ChatMessage {
 | 7 | message_type | `MessageType` | The type of message, different for one-to-one, public or group chats |
 | 8 | content_type | `ContentType` | The type of the content of the message | 
-| 9 | payload | `Sticker|Image|nil` | The payload of the message based on the content type |
+| 9 | payload | `Sticker` I `Image` I `EmojiReaction` I `EmojiReactionRetraction` I `nil` | The payload of the message based on the content type |
```
`|` would be parsed by the markdown renderer as a new table cell token, even though it was inside backticks. I've replaced `|` with `I` because markdown font renders `I` as "I"

oskarth commented:
Generally looks good to me, I'll leave this review for Status and core though :) @status-im/security-internal  might want to sanity too

rajeevgopalakrishna commented on docs/draft/6-payloads.md:
````diff
@@ -191,6 +199,49 @@ message ImageMessage {
+
+```protobuf
+message EmojiReactionRetraction {
+  string emoji_reaction_id = 1;
````
Does `emoji_reaction_id` refer to the corresponding `message_id` from `message EmojiReaction`?

Samyoul replied:
Yes.

So, `message EmojiReaction` will generate its own id and the `EmojiReactionRetraction` targets the id of the `EmojiReaction`. This approach will allow messages to have multiple reactions.

rajeevgopalakrishna reviewed (commented):
Looks good to me. Curious - can we apply the same retraction concept from here to messages as well?

oskarth commented:
Meta comment

> I've added these specs to the draft version of 6-payloads because I saw images in drafts and presumed that emoji reactions would also be considered on the same level or lower of spec stability.

Once it is live it should be marked as stable, because it is a contract with end users as well. I believe @cammellos has done some draft/stable splitting of specs in past to make this distinction clear, so that might be a a good place to look re that.

Samyoul commented:
@rajeevgopalakrishna 

>Looks good to me. Curious - can we apply the same retraction concept from here to messages as well?

We certainly could. In my mind this is the only way to "remove" messages from a p2p system. We need to send some kind of kill signal to peers that would honour the retraction of a message.

Samyoul commented:
@oskarth 

> Once it is live it should be marked as stable, because it is a contract with end users as well. I believe @cammellos has done some draft/stable splitting of specs in past to make this distinction clear, so that might be a a good place to look re that.

Thank you for that meta comment, this is useful information to know.

hesterbruikman commented:
Any changes based on the current status-go implementation @Samyoul ?

rasom reviewed (approved):
nice spec!

Samyoul committed:
Merge conflict resolved

Samyoul committed:
Moved MessageType protobuf into dedicated section

With the inclusion of emoji reactions the MessageType protobuf was split out from the ChatMessage protobuf so that the MessageType enums could be used in multiple protobuf message types.

Samyoul committed:
Merge branch 'master' into update/emoji-reactions
